openlibrary/plugins/upstream/code.py

```diff
@@ -191,7 +191,7 @@ def POST(self, key):
 
         user = accounts.get_current_user()
         is_admin = user and user.key in [m.key for m in web.ctx.site.get('/usergroup/admin').members]
-        if not (is_admin and web.ctx.site.can_write(key)):
+        if not (user.is_librarian() and is_admin and web.ctx.site.can_write(key)):
```

Comment:
I think this should be `if not ((user.is_librarian() or is_admin) and web.ctx.site.can_write(key)):`
to allow either admins or librarians to revert.

Reply:
Added proper check for librarians and users.

Reply:
Added proper check for librarians and users as well, thanks for pointing that out.